gallery/__init__.py

```diff
@@ -386,24 +390,25 @@ def add_file(file_name: str, path: str, dir_id: str, description: str, owner: st
     return file_model
 
 
-def refresh_thumbnail():
-    def refresh_thumbnail_helper(dir_model: Directory) -> str:
-        dir_children = [d for d in Directory.query.filter(Directory.parent == dir_model.id).all()]
-        file_children = [f for f in File.query.filter(File.parent == dir_model.id).all()]
-        for file in file_children:
-            if file.thumbnail_uuid != DEFAULT_THUMBNAIL_NAME:
-                return file.thumbnail_uuid
-        for d in dir_children:
-            if d.thumbnail_uuid != DEFAULT_THUMBNAIL_NAME:
-                return d.thumbnail_uuid
-        # WE HAVE TO GO DEEPER (inception noise)
-        for d in dir_children:
-            # TODO: Switch to iterative tree walk using a queue to avoid
-            # recursion issues with super large directory structures
-            return refresh_thumbnail_helper(d)
-        # No thumbnail found
-        return DEFAULT_THUMBNAIL_NAME
-
+def refresh_directory_thumbnail(dir_model: Directory) -> str:
+    dir_children = [d for d in Directory.query.filter(Directory.parent == dir_model.id).all()]
+    file_children = [f for f in File.query.filter(File.parent == dir_model.id).all()]
+    for file in file_children:
+        if file.thumbnail_uuid != DEFAULT_THUMBNAIL_NAME and not file.hidden:
+            return file.thumbnail_uuid
+    for d in dir_children:
+        if d.thumbnail_uuid != DEFAULT_THUMBNAIL_NAME:
+            return d.thumbnail_uuid
+    # WE HAVE TO GO DEEPER (inception noise)
+    for d in dir_children:
```

Comment:
Should we do this before we check through all the children?

Reply:
Don't think so, as currently the order it chooses a new thumbnail is:
```
if a child file has valid (non-hidden, non-default) thumbnail, use it
else if a child dir has a valid thumbnail, use it
else recurse through all child dirs to find valid thumbnails within them
```